go/vt/vttablet/tabletserver/tabletenv/config.go

```diff
@@ -138,8 +140,9 @@ func registerTabletEnvFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&currentConfig.TwoPCCoordinatorAddress, "twopc_coordinator_address", defaultConfig.TwoPCCoordinatorAddress, "address of the (VTGate) process(es) that will be used to notify of abandoned transactions.")
 	SecondsVar(fs, &currentConfig.TwoPCAbandonAge, "twopc_abandon_age", defaultConfig.TwoPCAbandonAge, "time in seconds. Any unresolved transaction older than this time will be sent to the coordinator to be resolved.")
 	flagutil.DualFormatBoolVar(fs, &currentConfig.EnableTxThrottler, "enable_tx_throttler", defaultConfig.EnableTxThrottler, "If true replication-lag-based throttling on transactions will be enabled.")
-	flagutil.DualFormatStringVar(fs, &currentConfig.TxThrottlerConfig, "tx_throttler_config", defaultConfig.TxThrottlerConfig, "The configuration of the transaction throttler as a text formatted throttlerdata.Configuration protocol buffer message")
+	flagutil.DualFormatStringVar(fs, &currentConfig.TxThrottlerConfig, "tx_throttler_config", defaultConfig.TxThrottlerConfig, "The configuration of the transaction throttler as a text formatted throttlerdata.Configuration protocol buffer message.")
 	flagutil.DualFormatStringListVar(fs, &currentConfig.TxThrottlerHealthCheckCells, "tx_throttler_healthcheck_cells", defaultConfig.TxThrottlerHealthCheckCells, "A comma-separated list of cells. Only tabletservers running in these cells will be monitored for replication lag by the transaction throttler.")
+	fs.Var((*topoproto.TabletTypeListFlag)(&currentConfig.TxThrottlerTabletTypes), "tx-throttler-tablet-types", "A comma-separated list of tablet types. Only tablets of this type are monitored for replication lag by the transaction throttler. Supported types are replica and/or rdonly.")
```

Comment:
My opinion: we should distance ourselves from proto-format flags. I understand that these are strongly typed, but on the other hand this isn't standard practice in `golang` world or outside `vitess`, and mostly these are difficult to craft by hand, what with all the escaping.

Personally I'd go with just a comma delimited `string` like here:

- https://github.com/vitessio/vitess/blob/11fd98da90c41bbfe522ec6761d731666978ab78/go/vt/vttablet/tabletserver/gc/tablegc.go#L67
- then split it like here: https://github.com/vitessio/vitess/blob/11fd98da90c41bbfe522ec6761d731666978ab78/go/vt/vttablet/onlineddl/executor.go#L326

Reply:
Not a hill to die on for now, as the current config is also already in proto format. Feel free to keep this as is. Possibly, in the future we may change this.

Reply:
Also going to tangent here: because we're using pflag, we can use `fs.StringSliceVar` and have pflag do the CSV splitting for us (and therefore move away from this "string var then we split it for you" code that's all over vitess)

Reply:
Docs: https://pkg.go.dev/github.com/spf13/pflag#FlagSet.StringSliceVar

Reply:
oh yes, definitely `StringSliceVar` over `string`. And I should do the same in existing codebase!